pkg/allocator/allocator.go

```diff
@@ -636,6 +636,15 @@ func (a *Allocator) Allocate(ctx context.Context, key AllocatorKey) (idpool.ID,
 		select {
 		case <-ctx.Done():
 			scopedLog.WithError(ctx.Err()).Warning("Ongoing key allocation has been cancelled")
+
+			// If the deadline as exceeded and there was a backend error, this
+			// means that kvstore connectivity is unstable. We fatal Cilium in
+			// hopes of finding a new backend.
+			if _, backendErr := a.backend.Status(); backendErr != nil &&
```

Comment:
I think we want this to trigger earlier. If we make it depends on `backendErr` it will require for all etcd endpoints to be unhealthy as a `2/3` endpoints situation will not trigger a `backendErr != nil` case.